lib/ldp/client/methods.rb

```diff
@@ -8,11 +8,24 @@ module Ldp::Client::Methods
   def initialize_http_client *http_client
```

Comment:
I'm a little scared of this signature (and, yes, it's at least partly my fault.) 

I'm not a fan of the 3-arg option you're proposing. Could we just yield the http client in `Ldp::Client#initialize` and let our downstream consumers do whatever?